txircd/ircd.py

```diff
@@ -470,10 +603,16 @@ def generateISupportList(self):
 		isupport = self.isupport_tokens.copy()
 		statusSymbolOrder = "".join([self.channelStatuses[status][0] for status in self.channelStatusOrder])
 		isupport["CHANMODES"] = ",".join(["".join(modes) for modes in self.channelModes])
+		isupport["CHANNELEN"] = self.config.get("channel_name_length", 64)
+		isupport["NETWORK"] = self.config["network_name"]
 		isupport["PREFIX"] = "({}){}".format("".join(self.channelStatusOrder), statusSymbolOrder)
 		isupport["STATUSMSG"] = statusSymbolOrder
 		isupport["USERMODES"] = ",".join(["".join(modes) for modes in self.userModes])
-		isupport["NETWORK"] = self.config["network_name"]
+
+		extraTokens = {}
+		self.runActionStandard("buildisupport", extraTokens)
+		isupport.update(extraTokens)
```

Comment:
If we're going to do this, why make a separate dict? We should
- pass the `isupport` dict if we don't care about the existing values
- if we want to enforce safety, use this dict to update the isupport dict in a way that doesn't overwrite preexisting values.